### PR TITLE
Add [GithubDeployments] badge (#4477)

### PR DESCRIPTION
**Pull request by TrueBrain**
My first attempt to add a badge here :) Few things worth mentioning:

- I could not find a good category, so I picked 'other'.
- I could not find a colouring scheme that fits deployments, so I created one.
- I tried to figure out how `*.tester.js` works, but the [documentation](https://github.com/badges/shields/blob/master/doc/service-tests.md) was not clear enough for me to do anything with (I still have no clue what `get()` is doing, and how to add anything to it). So I did not add any.

Btw, the [tutorial](https://github.com/badges/shields/blob/master/doc/TUTORIAL.md) is very awesome and well written; thank you for that.

Curious what you think of this!

In case you want to test how the badge looks, you can use https://github.com/OpenTTD/website ([environments](https://github.com/OpenTTD/website/deployments) 'staging' or 'production' works). It can be difficult to find a GitHub project with deployments enabled.

### CODE REVIEW

**Comment by chris48s** [comment]
I've also edited this PR title so that when you push service tests to the branch, they'll be run in CI

File: services/github/github-deployments.service.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'GitHub deployments',
+        namedParams: {
+          user: 'badges',
+          repo: 'shields',
+          environment: 'production',
```
Comment (calebcartwright) [inline services/github/github-deployments.service.js]: let's using the staging env instead so the example provides a better sample
```suggestion
          environment: 'shields-staging',
```

Comment (TrueBrain) [reply services/github/github-deployments.service.js]: Had not seen this repo also uses deployments :D Sweet, that makes the test-case also a lot easier, tnx :)

**Comment by TrueBrain** [comment]
@chris48s: thank you very much! Now adding the test was easy :) Rereading the page now also makes more sense. What totally put me on the wrong track, were the llinks to wercker itself. To me it read like we were calling their API, instead of the shields API. Totally makes sense now.

Related to the colouring: `renderBuildStatusBadge` has one drawback: it changes the message of the `greenStatus` always to `passing`. `passing` from a deployment point of view is a silly word to use. But I guess I can work around that relatively easy. At least avoids a ton of code duplication :)

File: services/github/github-deployments.service.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+const schema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      deployments: Joi.object({
+        nodes: Joi.array()
+          .items(
+            Joi.object({
+              latestStatus: Joi.object({
+                state: Joi.string(),
```
Comment (calebcartwright) [inline services/github/github-deployments.service.js]: We should use a stricter set of validations here based on the known/expected status values

File: services/github/github-deployments.service.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    })
+
+    let message = build_message
+
+    // renderBuildStatusBadge() does everything we need except it rewrites the
+    // message for status in 'greenStatuses' to 'passing'. This makes little
+    // sense in the context of a deployment. So rewrite it back to 'success'.
+    if (state == 'success') {
+      message = 'success'
+    }
```
Comment (calebcartwright) [inline services/github/github-deployments.service.js]: Given the above, is it really worth using the `renderBuildStatusBadge` helper here? Seems like all we'd be using it for here is to map a color to the `state` value from the deployment object

File: services/github/github-deployments.service.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    if (data.repository.deployments.nodes.length == 0) {
+      return { state: 'unknown' }
+    }
+
+    let state = data.repository.deployments.nodes[0].latestStatus.state
+    state = state.toLowerCase()
+
+    if (state == 'in_progress') {
+      return { state: 'in progress' }
+    }
```
Comment (calebcartwright) [inline services/github/github-deployments.service.js]: This (and the lowercasing) feel more like rendering responsibilities than transform responsibilities IMO. Generally speaking those are, respectively, presenting the data to the user in a pretty/friendly way, and extracting the relevant data portions from the raw response

File: services/github/github-deployments.service.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        }
+      `,
+      variables: { user, repo, environment },
+      schema,
+      transformErrors,
+    })
+  }
+
+  transform({ data }) {
+    if (data.repository.deployments.nodes.length == 0) {
```
Comment (calebcartwright) [inline services/github/github-deployments.service.js]: In what scenarios would this occur? Is it when the user specifies an invalid environment value, when there were no deployments to the specified environment, something else?

File: services/github/github-deployments.service.js (hunk 2 above)
Comment (calebcartwright) [reply services/github/github-deployments.service.js]: For reference, here's the raw values from GitHub that Shields will see (note the all upper case) https://developer.github.com/v4/enum/deploymentstatusstate/

File: services/github/github-deployments.service.js (hunk 5 above)
Comment (TrueBrain) [reply services/github/github-deployments.service.js]: That sounds a lot better :D One other corner case is when there is a deployment which did not receive a status update yet. I am not sure we can differentiate between "env not found" and "deployment without status". Also not sure it is worth the effort (given it is a rare corner case in normal operations).

**Comment by chris48s** [comment]
My initial though was that this maps well onto "Build Status" as most of the status values already existed there, but if we're trying to bash a square peg into a round hole, lets keep it custom :+1:

File: services/github/github-deployments.tester.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const t = (module.exports = require('../tester').createServiceTester())
+
+t.create('Deployments')
+  .get('/badges/shields/shields-staging.json')
+  .expectBadge({
+    label: 'state',
+    message: Joi.string(),
```
Comment (chris48s) [inline services/github/github-deployments.tester.js]: Can we define a more specific validation here.
For example, if the badge said `'environment not found'`, that would pass `Joi.string()`, but it shouldn't

Comment (TrueBrain) [reply services/github/github-deployments.tester.js]: I do apologize, as you mentioned that before. For some reason .. I forgot again :) Fixed!

**Comment by calebcartwright** [comment]
Awesome, thanks @TrueBrain!

**Pull request merged**